### PR TITLE
feat(router): verify service for applepay merchant registration

### PR DESCRIPTION
**Pull request by prajjwalkumar17**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

New verify flow for applepay merchant verification

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/api_models/src/verifications.rs
[the remark this answers is not in the document]
Comment (jarnura) [reply crates/api_models/src/verifications.rs:1]: add support to generate open_api_spec also.

File: crates/router/src/utils/verification.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    let applepay_response =
+        response.change_context(api_error_response::ApiErrorResponse::InternalServerError)?;
+    // Error is already logged
+    Ok(match applepay_response {
+        Ok(_) => services::api::ApplicationResponse::Json(ApplepayMerchantResponse {
+            status_code: "200".to_string(),
+            status_message: "Applepay verification Completed".to_string(),
+        }),
+        Err(_) => services::api::ApplicationResponse::Json(ApplepayMerchantResponse {
```
Comment (jarnura) [inline crates/router/src/utils/verification.rs:82]: log the error, don't discard the errors

**Review by jarnura — CHANGES_REQUESTED** [review]

File: crates/api_models/src/verifications.rs
[the remark this answers is not in the document]
Comment (prajjwalkumar17) [reply crates/api_models/src/verifications.rs:1]: As decided, due to this route being an internal route as of now we are not proceeding with adding it to openapi_spec

File: crates/router/src/lib.rs [hunk 2]
```diff
@@ -138,6 +138,11 @@ pub fn mk_app(
             .service(routes::Disputes::server(state.clone()))
     }
 
+    #[cfg(feature = "release")]
```
Comment (jarnura) [inline crates/router/src/lib.rs:141]: just add olap

**Review by jarnura — CHANGES_REQUESTED** [review]

File: crates/router/src/lib.rs (hunk 2 above)
Comment (prajjwalkumar17) [reply crates/router/src/lib.rs]: as discussed added all(kms, olap)

**Review by jarnura — DISMISSED** [review]

File: crates/router/src/utils/verification.rs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+use crate::{
+    core::errors::{self, api_error_response},
+    headers, logger,
+    routes::AppState,
+    services, types, utils,
+};
+
+const APPLEPAY_MERCHANT_VERIFICATION_URL: &str =
+    "https://apple-pay-gateway.apple.com/paymentservices/registerMerchant";
+const APPLEPAY_INTERNAL_MERCHANT_IDENTIFIER: &str = "merchant.com.noon.juspay";
```
Comment (Sangamesh26) [inline crates/router/src/utils/verification.rs:17]: Can you check, is `APPLEPAY_INTERNAL_MERCHANT_IDENTIFIER` requires KMS encryption?

Comment (prajjwalkumar17) [reply crates/router/src/utils/verification.rs:17]: That wouldn't require it as anyways it is an internal api and will be helpfull as it is for debugging purposes

**Review by Sangamesh26 — APPROVED** [review]
LGTM :)

**Review by jarnura — APPROVED** [review]

File: crates/router/src/utils/verification.rs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        .await
+        .decrypt(encrypted_cert)
+        .await
+        .change_context(api_error_response::ApiErrorResponse::InternalServerError)?;
+
+    let key_data = kms::get_kms_client(kms_config)
+        .await
+        .decrypt(encrypted_key)
+        .await
+        .change_context(api_error_response::ApiErrorResponse::InternalServerError)?;
```
Comment (NishantJoshi00) [inline crates/router/src/utils/verification.rs:51]: Can we have this decrypted similar to all the other KMS decryptions that are being done? in `kms.rs`

**Pull request merged**